### ISSUE TITLE
AWS Amplify Network Issue on Watch OS development

### ISSUE THREAD

**Issue by pravarsha21** [aws-amplify/amplify-swift#3215]
### Describe the bug

I have created a voice recorder application for watch only app in xcode. The logic is when the user stops the recording, the recorded audio file is transferred red to AWS S3 bucket.

The code is working fine when I run the app on Xcode Watch OS simulator.

But when I install it on the physical watch, every time  it is giving network error. I have added all the required permissions and also pinged a basic website and it returned as OK.

### Steps To Reproduce

```swift
I think it is difficult to reproduce since you need the code of my app
```

### Expected behavior

The expected behaviour is it should transfer the files to S3

### Amplify Framework Version

2.17.1

### Amplify Categories

Storage

### Dependency manager

Swift PM

### Swift version

5.8

### CLI version

12.4.0

### Xcode version

14+

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


🔍 File exists, proceeding with upload.
🔍 Network available: true
🔍 Uploading Recording_09-10-2023_22-51-23_966
🔍 Data size: 43304 bytes
2023-09-10T22:51:34-0500 info CognitoIdentityClient : [Logging] Request: POST https:443 
 Path: / 
 Content-Type: application/x-amz-json-1.1, 
User-Agent: aws-sdk-swift/1.0 api/cognito-identity/1.0 os/watchOS/9.6.1 lang/swift/5.8 lib/amplify-swift/2.17.1, 
Content-Length: 79, 
x-amz-user-agent: aws-sdk-swift/1.0, 
Host: cognito-identity.us-east-2.amazonaws.com, 
X-Amz-Target: AWSCognitoIdentityService.GetId 
 Optional([])
2023-09-10T22:51:34-0500 info SerialExecutor : [Logging] Creating connection pool for Optional("https://cognito-identity.us-east-2.amazonaws.com/?")with max connections: 50
🛑 Full error object: StorageError: The operation couldn’t be completed. (AwsCommonRuntimeKit.CommonRunTimeError error 0.)
Recovery suggestion: Check your network connection, retry when the network is available.
HTTP Response stauts code: nil
🛑 Storage error: The operation couldn’t be completed. (AwsCommonRuntimeKit.CommonRunTimeError error 0.). Check your network connection, retry when the network is available.
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

_No response_

### Platforms

watchOS

### OS Version

16

### Device

Apple Watch 8

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by phantumcode** [comment]
@pravarsha21 Thanks for opening the issue. We will investigate and get back to you soon.

**Comment by ruisebas** [comment]
@pravarsha21 could you please share the logs with `verbose` logging enabled? 
They might include additional useful information.
```swift
Amplify.Logging.logLevel = .verbose
```

Also, are you using the `uploadFile(key:local:)` API and doing something like this?
```swift
let file = // your recording file
let uploadTask = Amplify.Storage.uploadFile(
    key: "yourKey",
    local: file
)

_ = try await uploadTask.value
```

**Comment by atierian** [comment]
Hey @pravarsha21, quick update from our side - we've identified the issue and are working on a fix.
We'll let you know once it's released.

**Comment by atierian** [comment]
This fix was released with [2.18.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.18.1)

Thanks again for reporting it @pravarsha21

**Comment by SrinivasKalagotla** [comment]
Thankyou for solving the issue, it was very helpful for me and my team mate Pravarsha.

**Comment by pravarsha21** [comment]
thankyou @atierian for solving the issue .

**Comment by 5d** [comment]
Hi @gurupatel107,  

To better assist you, could you please provide more details on the difficulties or constraints you are facing with upgrading the minimum watchOS support to version 9? Understanding the specific challenges will help us explore potential solutions or consider adjustments in our future planning.

**Comment by gurupatel107** [comment]
Sure @5d.

Our existing application for watchOS is already on App Store and support minimum watchOs version 7.6. 
Recent changes we had done were adding AWS authentication for iOS app and had to add same in watch app too using this library. 

As some users may still be utilising watchOS versions 7 or 8. Given our commitment to maintaining compatibility across different versions, we are faced with the challenge of ensuring that our Watch app continues to support a broad range of watchOS versions. Consequently, setting the minimum deployment version to 9+ is not currently feasible in alignment with our internal guidelines and decisions.

So we would like to have help if any possible update were version 2.17.2 (last version which support watchOs 7+) with the current bug fix of including this [PR#3230](https://github.com/aws-amplify/amplify-swift/pull/3230). 

Looking forward to have your collaboration in this situation.

**Comment by atierian** [comment]
Hey @gurupatel107. Apologies for the delayed response here; we missed this because the issue was closed.

Amplify Swift supports watchOS 9 as a minimum target. However, there is nothing (currently) that requires watchOS 9 in Amplify Swift. To workaround this, you can add Amplify Swift as a [local package dependency](https://developer.apple.com/documentation/xcode/editing-a-package-dependency-as-a-local-package) and change [this line](https://github.com/aws-amplify/amplify-swift/blob/0b1a2da28c0d3b7a64ac6d72eb4a85eb1885ea17/Package.swift#L9) in the Package.swift to `.watchOS(.v7)`.

**Please note** that we can't make any guarantees that APIs requiring watchOS 9 won't be used in the future -- in other words, this workaround may not work for future versions.

I understand that every app has its own requirements around minimum supported versions. In case you haven't already, I'd encourage you to consider that the most recent Apple Watch unable to run watchOS 9 is the Series 3 from 2017 [[source](https://iosref.com/watchos)].

If you need any further assistance on this, please open a new issue. Thanks for your patience!
